tests/envs/pettingzoo/test_pettingzoo.py

```diff
@@ -0,0 +1,64 @@
+import shutil
+from os.path import isdir
+
+import pytest
+
+from sample_factory.algo.utils.context import reset_global_context
+from sample_factory.algo.utils.misc import ExperimentStatus
+from sample_factory.train import run_rl
+from sample_factory.utils.utils import log
+from sf_examples.train_pettingzoo_env import make_pettingzoo_env, parse_custom_args, register_custom_components
+from tests.envs.utils import eval_env_performance
+from tests.utils import clean_test_dir
+
+
+class TestPettingZooEnv:
+    @pytest.fixture(scope="class", autouse=True)
+    def register_pettingzoo_fixture(self):
+        register_custom_components()
+        yield  # this is where the actual test happens
+        reset_global_context()
```

Comment:
Added this (similar to other env tests)
this resets global encoder factory, otherwise other tests can fail if this runs first.

(I'd be the first to admit this is not a perfect solution but hey I wrote this years ago)